linera-rpc/src/node_provider.rs

```diff
@@ -32,14 +39,22 @@ impl ValidatorNodeProvider for NodeProvider {
 
     fn make_node(&self, address: &str) -> anyhow::Result<Self::Node, NodeError> {
         let address = address.to_lowercase();
+        let mut pool = self.client_pool.lock().unwrap();
+        if let Some(client) = pool.get(&address) {
+            return Ok(client.clone());
```

Comment:
Do both clients handle reconnecting if the other side dropped the connection?

Reply:
Exactly. This needs to be tested